sb.c

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2017 Ryan Armstrong <ryan@cavaliercoder.com>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*
+ * C String Builder - https://github.com/cavaliercoder/c-stringbuilder
+ *
+ * sb.c is a simple, non-thread safe String Builder that makes use of a
+ * dynamically-allocated linked-list to enable linear time appending and
+ * concatenation.
+ */
+#include <stdlib.h>
+#include <stdio.h>
+#include <stdarg.h>
+#include <string.h>
+#include "sb.h"
+
+/*
+ * sb_create returns a pointer to a new stringbuilder or NULL if memory is not
+ * available.
+ */
+stringbuilder *sb_create()
+{
+	stringbuilder *sb = (stringbuilder*) calloc(sizeof(stringbuilder), 1);
+	return sb;
+}
+
+/*
+ * sb_append adds a copy of the given string to a stringbuilder.
+ */
+int sb_append(stringbuilder *sb, const char *str)
+{
+	if (NULL == sb)
+		return SB_FAILURE;
+
+	int				length = 0;
+	stringfragment	*frag = NULL;
+
+	if (NULL == str || '\0' == *str)
+		return sb->length;
+
+	length = strlen(str);
+	frag = (stringfragment*) malloc(sizeof(stringfragment) + (sizeof(char) * length));
```

Comment:
Can you please remove all casts from allocation functions. Something we don't do at all. :)